stubs/AMQPChannel.php

```diff
@@ -179,4 +179,69 @@ public function getConnection()
     public function basicRecover($requeue = true)
     {
     }
+
+    /**
+     * Set the channel to use publisher acknowledgements. This can only used on a non-transactional channel.
+     */
+    public function confirmSelect()
+    {
+    }
+
+    /**
+     * Set callback to process basic.ack and basic.nac AMQP server methods (applicable when channel in confirm mode).
+     *
+     * @param callable|null $ack_callback
+     * @param callable|null $nack_callback
+     *
+     * Callback functions with all arguments have the following signature:
+     *
+     *      function ack_callback(int $delivery_tag, bool $multiple) : bool;
+     *      function nack_callback(int $delivery_tag, bool $multiple, bool $requeue) : bool;
+     *
+     * and should return boolean false when wait loop should be canceled.
+     *
+     * Note, basic.nack server method will only be delivered if an internal error occurs in the Erlang process
+     * responsible for a queue (see https://www.rabbitmq.com/confirms.html for details).
+     *
+     */
+    public function setConfirmCallback(callable $ack_callback=null, callable $nack_callback=null)
+    {
+    }
+
+    /**
+     * Wait until all messages published since the last call have been either ack'd or nack'd by the broker.
+     *
+     * Note, this method also catch all basic.return message from server.
+     */
+    public function waitForConfirm($timeout = 0.0)
```

Comment:
Microseconds can be passed as a float part (after dot). Do we need to be microsec based? Overall precision is not good at all no, e.g. you can start waiting for 1.5 and it can be even 2.0 sec or smth, small values are not very precise, so I don't think we'll have any benefit from moving to microsec.